### PR TITLE
Bug 1848512: Wrong interpretation of labels for resource quota diagram

### PR DESCRIPTION
**Pull request by dtaylor113**
Needed to provide `data.x` for chart labels/tooltip

### Before
image

### After
image

### CODE REVIEW

File: frontend/public/components/resource-quota.jsx [hunk 1]
```diff
@@ -191,16 +191,32 @@ const NoQuotaGuage = ({ title, className }) => (
 
 export const QuotaGaugeCharts = ({ quota, resourceTypes, chartClassName = null }) => {
   const resourceTypesSet = new Set(resourceTypes);
+  const cpuRequestUsagePercent =
```
Comment (jcaianirh) [inline frontend/public/components/resource-quota.jsx:194]:
```suggestion
  const cpuRequestUsagePercent =getResourceUsage(
    quota,
    resourceTypesSet.has('requests.cpu') ? 'requests.cpu' : 'cpu',
  ).percent;
```

File: frontend/public/components/resource-quota.jsx [hunk 2]
```diff
@@ -191,16 +191,32 @@ const NoQuotaGuage = ({ title, className }) => (
...
+    ? getResourceUsage(quota, 'limits.cpu').percent
+    : 0;
+  const memoryRequestUsagePercent =
+    resourceTypesSet.has('requests.memory') || resourceTypesSet.has('memory')
+      ? getResourceUsage(
+          quota,
+          resourceTypesSet.has('requests.memory') ? 'requests.memory' : 'memory',
+        ).percent
+      : 0;
+  const memoryLimitUsagePercent = resourceTypesSet.has('limits.memory')
```
Comment (jcaianirh) [inline frontend/public/components/resource-quota.jsx:209]:
```suggestion
  const memoryLimitUsagePercent = getResourceUsage(quota, 'limits.memory').percent;
```

File: frontend/public/components/resource-quota.jsx (hunk 1 above)
Comment (dtaylor113) [reply frontend/public/components/resource-quota.jsx]: fixed

File: frontend/public/components/resource-quota.jsx (hunk 2 above)
Comment (dtaylor113) [reply frontend/public/components/resource-quota.jsx]: fixed

**Comment by dtaylor113** [comment]
Hi @jcaianirh, applied that patter to all 4 lines.   Had to revert the check between showing a `GaugeChart` chart or `NoQuotaGuage` chart, in order to actually display a `0%` value, which is the state when a resource quota is created but no pods applied yet.

**Comment by dtaylor113** [comment]
Hi @spadgett, @kyoto, and @rhamilto,  looking for an approval in order to merge this PR.  -thanks :-)

File: frontend/public/components/resource-quota.jsx [hunk 3]
```diff
@@ -191,16 +191,24 @@ const NoQuotaGuage = ({ title, className }) => (
...
       {resourceTypesSet.has('requests.cpu') || resourceTypesSet.has('cpu') ? (
         <div className="co-resource-quota-gauge-chart">
           <GaugeChart
             data={{
-              y: getResourceUsage(
-                quota,
-                resourceTypesSet.has('requests.cpu') ? 'requests.cpu' : 'cpu',
-              ).percent,
+              x: `${cpuRequestUsagePercent}%`,
+              y: cpuRequestUsagePercent,
```
Comment (spadgett) [inline frontend/public/components/resource-quota.jsx:211]: Is there a reason we append `%` for `x` and not for `y`?

Comment (dtaylor113) [reply frontend/public/components/resource-quota.jsx:211]: `y' is the actual numeric data, `x` is the category string/label, which is used in the tooltip.  
https://formidable.com/open-source/victory/docs/victory-pie/

**Pull request merged**